src/humanPose/MuriLens.js

```diff
@@ -0,0 +1,101 @@
+import {MotionStudyLens} from "./MotionStudyLens.js";
+import {MuriScore} from "./MuriScore.js";
+import {MotionStudyColors} from "./MotionStudyColors.js";
+import {ERGO_ANGLES, ERGO_OFFSETS} from "./constants.js";
+
+/**
+ * MuriLens is a lens that calculates the Muri score for each pose in the history. Individual components of the score
+ * are visualised on relevant joints/bones.
+ */
+export class MuriLens extends MotionStudyLens {
+    /**
+     * Creates a new MuriLens object.
+     */
+    constructor() {
+        super("Muri Ergonomics");
+
+        this.muriScore = new MuriScore();
+    }
+    
+    applyLensToPose(pose, force = false) {
+        if (!force && pose.metadata.muriScores) {
+            return false;
+        } 
+        const ergonomicsData = this.muriScore.calculateForPose(pose);
+        pose.forEachJoint(joint => {
+            joint.muriScore = ergonomicsData.jointScores[joint.name];
+            joint.muriColor = ergonomicsData.jointColors[joint.name];
+        });
+        pose.forEachBone(bone => {
+            bone.muriScore = ergonomicsData.boneScores[bone.name];
+            bone.muriColor = ergonomicsData.boneColors[bone.name];
+        });
+
+        // populate pose metadata with base ergonomic data
+        pose.metadata.ergonomics = {};
+        Object.values(ERGO_ANGLES).forEach(angleName => {
+            // angles which could not be calculated are explicitly stored as null
+            if (ergonomicsData.angles.hasOwnProperty(angleName)) {
+                pose.metadata.ergonomics[angleName] = ergonomicsData.angles[angleName];
+            }
+            else {
+                pose.metadata.ergonomics[angleName] = null;
+            }
+        });
+        // TODO?: Volvo has bit different definition of this angle (straight arm has 180 deg). However, this definition is not properly established and is inconsistent with other angles which are zero in the netural pose. 
```

Comment:
What would you recommend for this moving forward? Should we have some mechanism for Volvo to use their definition while keeping it as 0-based for everyone else?

Reply:
For Volvo, this is a new strain type which they introduced recently, so it's definition is not well established. Also, I think they are flexible to some extent to adjust their Muri analysis. So my recommendation is to suggest them to adopt our definition.
Moreover, Valentin wants to keep most of the data within Toolbox based on the current direction. So these raw angle values may not be exposed at all. So as long as Volvo sees the right scores/colors for their angle ranges in our UI, it does not matter what exactly is the underlying value (angle or (180 - angle) )